lib/usbd/class/usbd_msc.c

```diff
@@ -275,6 +276,30 @@ static void scsi_read_capacity(usbd_msc *ms,
 	}
 }
 
+
+
+static 
+void scsi_read_format_capacities(usbd_msc *ms,
+ 			       struct usb_msc_trans *trans,
+ 			       enum trans_event event)
+{
+	if (EVENT_CBW_VALID == event) {
+		usb_msc_rfc_capacity_list_header*  h      = (usb_msc_rfc_capacity_list_header*)trans->msd_buf;
+		usb_msc_rfc_capacity_descriptor*   descr  = (usb_msc_rfc_capacity_descriptor*) (trans->msd_buf+sizeof(*h));
+		
+		u24_assign(h->dummy, 0);
+		h->list_len = sizeof(*descr);
+
+		msc_lba_t size = usbd_msc_blocks(ms->backend);
+		descr->blocks_count = HTONL(size);
+		descr->code 		= rfc_dc_Fomatted;
+		u24_assign(descr->block_size, USBD_MSC_SEC_SIZE);
+
+		trans->bytes_to_send = sizeof(*h) + h->list_len;
+		set_sbc_status_good(ms);
+	}
+}
+
```

Comment:
What are you trying to do with the new code.
please explain the purpose.
a comment or commit msg should be ok.